### ISSUE TITLE
Use CSG to create collision shapes

### ISSUE THREAD

**Issue by BastiaanOlij** [godotengine/godot#19933]
Now that I'm procedurally generating a bridge using the new CSG logic, would be nice to use the mesh building logic to create collision shapes.

**Comment by nobuyukinyuu** [comment]
I was hoping to see this idea applied to polygon2Ds for a while but it's not directly related to CSG;  it *is*, however, related to the scope of what this thread is requesting.  2d NavigationMeshes already are able to clip themselves using `xor` in a manner that is analogous to what CSG does in 3d, but no clipping operations (save for perhaps `or`, in a functional capacity) are available for collision polygons in 2d.  This situation is similar to CSG being unable to be used to create collision shapes in 3d; supporting both would bring relative feature parity.

Separate request?  Similar scope?  IDK, just thought I'd throw that out there.

**Comment by BastiaanOlij** [comment]
@nobuyukinyuu I'm afraid I can't say too much about the 2D implications to this, would be an entirely different set of operations I think. Best to raise a separate issue I think.

As far as collision shapes for CSG go, there is a tickbox on the CSG that I thought didn't do what I wanted it to do but kinda does. 

Tick "use collision" and it will add a static collision shape. 

Unfortunately it is always a static shape directly added to the physics engine and does not work with RigidBody or KinematicBody, but for my needs it suffices.

**Comment by akien-mga** [comment]
Closed by mistake when rebasing?

**Comment by BastiaanOlij** [comment]
@akien-mga no, its already implemented, there is a "use collision" tickbox on all CSG nodes that turn them into a static body with collision shape.
